src/app/courses/add-course/add-course.component.ts

```diff
@@ -1,63 +1,94 @@
 import { CoursesService } from 'src/app/shared/services/course-service/courses.service';
 
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
-import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators, FormGroupDirective, NgForm } from '@angular/forms';
 import { Router } from '@angular/router';
 import { Store } from '@ngrx/store';
+import { ErrorStateMatcher } from '@angular/material/core';
 
 import { AuthorizationService } from '../../shared/services/auth-service/auth-service';
+import { AuthorsService } from '../../shared/services/authors.service';
 import { LoadingService } from '../../shared/services/loading.service';
 import * as fromStore from '../../store/reducers';
-import {AddCourseFail, AddCourseSuccess} from '../../store/actions/courses.action';
+import { AddCourseFail, AddCourseSuccess } from '../../store/actions/courses.action';
+
+
+export class MyErrorStateMatcher implements ErrorStateMatcher {
+    isErrorState(control: FormControl | null, form: FormGroupDirective | NgForm | null): boolean {
+      const isSubmitted = form && form.submitted;
+      return !!(control && control.invalid && (control.dirty || control.touched || isSubmitted));
+    }
+  }
 
 @Component({
-  selector: 'app-add-course',
-  templateUrl: './add-course.component.html',
-  styleUrls: ['./add-course.component.scss']
+    selector: 'app-add-course',
+    templateUrl: './add-course.component.html',
+    styleUrls: ['./add-course.component.scss']
 })
 export class AddCourseComponent implements OnInit {
-  public itemId: number;
-  public coursesForm: FormGroup;
-  public isAuth: boolean;
-
-  constructor(
-    private _coursesService: CoursesService,
-    private _authService: AuthorizationService,
-    private fb: FormBuilder,
-    private router: Router,
-    private loadingService: LoadingService,
-    private store: Store<fromStore.State>,
-    ) { }
-
-  ngOnInit(){
-    this.coursesForm = this.fb.group({
-      title: [''],
-      description: [''],
-      creationDate: [''],
-      duration: ['']
-    });
-
-    this.isAuth = this._authService.isAuthenticated;
-    this.loadingService.hide();
+    public coursesForm: FormGroup;
+    public isAuth: boolean;
 
-  }
+    constructor(
+        private _coursesService: CoursesService,
```

Comment:
the services are having different naming conventions, either they are with underscore "_" or without.